### PR TITLE
refactor: billing UI improvements and error messages

### PR DESCRIPTION
**Pull request by amosmachora**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
In this PR I have fixed an issue that caused the dialog to dissapear momentarily when the request has been made. Also I have added a better way of indicating a loading phone number.

Also I have added a better error message `An error occurred when initiating stk push` over the alternative `Failed to fetch` which i felt was not that clear.

## Video

https://github.com/user-attachments/assets/c090f877-14d7-4b39-9a93-0c2da936fb0b

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Review by donaldkibet — APPROVED** [review]
LGTM Thanks @amosmachora.

File: packages/esm-billing-app/src/config-schema.ts [hunk 1]
```diff
@@ -100,31 +95,6 @@ export const configSchema = {
     _default: true,
     _description: 'Whether to enforce bill payment or not for patient to receive service',
   },
-  mpesaCallbackUrl: {
```
Comment (ojwanganto) [inline packages/esm-billing-app/src/config-schema.ts:103]: Please confirm that these details were moved to the mpesa aggregator app

File: packages/esm-billing-app/src/m-pesa/mpesa-resource.tsx [hunk 2]
```diff
@@ -26,19 +26,27 @@ export const initiateStkPush = async (
       }),
     });
 
-    if (!res.ok && res.status === 403) {
-      const error = new Error('Health facility M-PESA data not configured.');
-      throw error;
+    if (res.ok) {
+      const response: { requestId: string } = await res.json();
+      setNotification({ message: 'STK Push sent successfully', type: 'success' });
```
Comment (ojwanganto) [inline packages/esm-billing-app/src/m-pesa/mpesa-resource.tsx:31]: Should we also translate the success text?

File: packages/esm-billing-app/src/config-schema.ts (hunk 1 above)
Comment (amosmachora) [reply packages/esm-billing-app/src/config-schema.ts:103]: Yes they were. Everything Mpesa related is handled from the aggregator. The only thing we need here mpesa related is the aggregator base url.

File: packages/esm-billing-app/src/m-pesa/mpesa-resource.tsx (hunk 2 above)
Comment (amosmachora) [reply packages/esm-billing-app/src/m-pesa/mpesa-resource.tsx:31]: I'm still grasping this translations concept to be honest. What do you think @donaldkibet ?

**Comment by ojwanganto** [comment]
I will just merge it in

**Pull request merged**